# Q1

1. 

Let $X_i$ be the number of deluxe and professional bicycle produced.

objective: Max $Z=10X_1+15X_2$

s.t.

$2X_1+4X_2\leq100$

$3X_1+2X_2 \leq 80$

$X_1,X_2 \geq 0$

In [26]:
import gurobipy as grb

In [2]:
A1 = grb.Model('Q1')

Using license file /Users/chelseayeung/gurobi.lic
Academic license - for non-commercial use only


In [3]:
X={}
X[1]=A1.addVar(obj=10,name="deluxe")
X[2]=A1.addVar(obj=15,name="professional")

In [4]:
A1.ModelSense = grb.GRB.MAXIMIZE

In [7]:
A1.addConstr(2*X[1]+4*X[2]<=100) # al
A1.addConstr(3*X[1]+2*X[2]<=80) # steel

<gurobi.Constr *Awaiting Model Update*>

In [8]:
A1.write('Tut2Q1.lp')

In [9]:
A1.optimize()

Gurobi Optimizer version 9.0.3 build v9.0.3rc0 (mac64)
Optimize a model with 2 rows, 2 columns and 4 nonzeros
Model fingerprint: 0x21151285
Coefficient statistics:
  Matrix range     [2e+00, 4e+00]
  Objective range  [1e+01, 2e+01]
  Bounds range     [0e+00, 0e+00]
  RHS range        [8e+01, 1e+02]
Presolve time: 0.01s
Presolved: 2 rows, 2 columns, 4 nonzeros

Iteration    Objective       Primal Inf.    Dual Inf.      Time
       0    2.5000000e+31   2.750000e+30   2.500000e+01      0s
       2    4.1250000e+02   0.000000e+00   0.000000e+00      0s

Solved in 2 iterations and 0.01 seconds
Optimal objective  4.125000000e+02


In [16]:
print(X[1].x)
print(X[2].x)
print("objective value: 412.5")

15.0
17.5
objective value: 412.5


# Sensitivity report

In [20]:
#Excel equivalent: Reduced Cost
print ('---------------------------------------------')
print ('Reduced Cost:')
for i in A1.VarName:
        ind = A1.VarName.index(i)
        print ('Variable - {} = {:.2f}'.format(i,A1.RC[ind]))
print ('---------------------------------------------')
#Excel equivalent: ObjCoef + Allowable increase
print ('Largest value of objective coefficient at which the current optimal basis would remain optimal:')
for i in A1.VarName:
        ind = A1.VarName.index(i)
        print ('Variable - {} = {:.2f}'.format(i,A1.SAObjUp[ind]))
print ('---------------------------------------------')
#Excel equivalent: ObjCoef - Allowable decrease
print ('Smallest value of objective coefficient at which the current optimal basis would remain optimal:')
for i in A1.VarName:
        ind = A1.VarName.index(i)
        print ('Variable - {} = {:.2f}'.format(i,A1.SAObjLow[ind]))
print ('---------------------------------------------')
print ('---------------------------------------------')
#Excel equivalent: Shadow Prices
print ('Dual Variables/Shadow prices:')
for i in A1.ConstrName:
        ind = A1.ConstrName.index(i)
        print ('Constraint - {} = {:.3f}'.format(i,A1.Pi[ind]))
print ('---------------------------------------------')
#Excel equivalent: Contraint RHS - Allowable Decrease
print ('Smallest RHS value at which the current optimal basis would remain optimal:')
for i in A1.ConstrName:
        ind = A1.ConstrName.index(i)
        print ('Constraint - {} = {:.2f}'.format(i,A1.SARHSLow[ind]))
print ('---------------------------------------------')
#Excel equivalent: Constraint RHS + Allowable increase
print ('largest RHS value at which the current optimal basis would remain optimal:')
for i in A1.ConstrName:
        ind = A1.ConstrName.index(i)
        print ('Constraint - {} = {:.2f}'.format(i,A1.SARHSUp[ind]))

---------------------------------------------
Reduced Cost:
Variable - deluxe = 0.00
Variable - professional = 0.00
---------------------------------------------
Largest value of objective coefficient at which the current optimal basis would remain optimal:
Variable - deluxe = 22.50
Variable - professional = 20.00
---------------------------------------------
Smallest value of objective coefficient at which the current optimal basis would remain optimal:
Variable - deluxe = 7.50
Variable - professional = 6.67
---------------------------------------------
---------------------------------------------
Dual Variables/Shadow prices:
Constraint - R0 = 3.125
Constraint - R1 = 1.250
---------------------------------------------
Smallest RHS value at which the current optimal basis would remain optimal:
Constraint - R0 = 53.33
Constraint - R1 = 50.00
---------------------------------------------
largest RHS value at which the current optimal basis would remain optimal:
Constraint - R0 = 160.00

In [21]:
15*29+17.5*15

697.5

(3)The coefficient of $X_1$ in objective function increases by 10 which is smaller than the allowable increase of 12.5. The solution will still be optimal. The objective value will be $15\times 20+17.5\times15 = 697.5$

(4)The allowable decrease for deluxe frames in objective function is 2.5. The decrease of profit is 4 which is greater than the allowable decrease. The solution may not be optimal.

(5)The sum of changes percentages is 0.88 which is smaller than 1. According to 100% rule, the solution is still optimal.

In [25]:
(16-10)/(22.5-10)+(17-15)/(20-15)

0.88

(6)$p \geq 3\times 3.125 + 3\times 1.25$, the profit should be greater than 13.125.

In [23]:
3*3.125+3*1.25

13.125